tests/test_postproc.py

```diff
@@ -0,0 +1,56 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+tests.test_postproc- Tests of various functions in newspack, among which rar_renamer()
+"""
+
+from sabnzbd.postproc import *
+from unittest import mock
+import os, shutil
+
+
+class TestPostProc:
+
+    # Tests of rar_renamer() (=deobfuscate) against various input directories
+    def test_rar_renamer(self):
+
+        # Function to deobfuscate one directorty with rar_renamer()
+        def deobfuscate_dir(sourcedir):
+            # sourcedir is the relative path to the directory with obfuscated files
+
+            # enrich to absolute path:
+            sourcedir = os.path.join(os.getcwd(), sourcedir)
+            # We create a workingdir, because the filenames are really changed
+            workingdir = sourcedir + "---WORKING-DIR"
+            # if workingdir still there from previous run, remove it:
+            try:
+                shutil.rmtree(workingdir)
+            except:
+                pass
+            # create a fresh copy
+            try:
+                shutil.copytree(sourcedir, workingdir)
+            except:
+                assert False
+
+            # And now let the magic happen:
+            nzo = mock.Mock()
+            nzo.final_name = "some-download-name"
+            number_renamed_files = rar_renamer(nzo, workingdir)
+
+            # Remove workingdir again
```

Comment:
If here we could verify the exact names of the renamed files on the filesystem, it would make this test 100% perfect 🥇 
I would suggest a simple:
```python
expected_files = ['bla.part01.rar', 'bla.part02.rar']
files_renamed = os.listdir('output_folder')
for test_file in expected_files:
    assert test_file in files_renamed 
```
The `expected_files` can be an input to your `deobfuscate_dir` function.

Reply:
Beautiful, but dangerous: in case of >1 rarsets, it is non-deterministic aka non-predictable which rarset gets which prefix ("A", or "B", or ...) 
And IMHO the `expected_files` would become big.

Alternatives:
* Check if there is one `*part009.rar` and two `*part007.rar`. That is a very strong indication rar_renamer() has correctly reached the end of both rar sets.
* change the prefix method rar_renamer() itself: base prefix on the biggest file inside the first rar (could be meaningful in regular usage / debugging), or a hash of it. But this would make rar_renamer() even more complex (if we take into account edge cases), and it's already complex now.


```
A--some-download-name.part001.rar
A--some-download-name.part002.rar
A--some-download-name.part003.rar
A--some-download-name.part004.rar
A--some-download-name.part005.rar
A--some-download-name.part006.rar
A--some-download-name.part007.rar
A--some-download-name.part008.rar
A--some-download-name.part009.rar
B--some-download-name.part001.rar
B--some-download-name.part002.rar
B--some-download-name.part003.rar
B--some-download-name.part004.rar
B--some-download-name.part005.rar
B--some-download-name.part006.rar
B--some-download-name.part007.rar
```